h2o-core/src/main/java/water/util/MathUtils.java

```diff
@@ -677,4 +677,13 @@ public static int compareUnsigned(long hiA, long loA, long hiB, long loB) {
     int resLo = compareUnsigned(loA, loB);
     return resHi != 0 ? resHi : resLo;
   }
+
+  /**
+   * Logloss
+   * @param err prediction error (between 0 and 1)
+   * @return logloss
+   */
+  public static double logloss(double err) {
+    return -Math.log(Math.max(1e-15, 1.0-err));
+  }
```

Comment:
It may be more efficient to have
  return Math.min(-Math.log(1.00-err), Aconstant)

where AConstant = -Math.log(1e-15)